### PR TITLE
Store tokens 5 seconds shorter than expiry time to avoid using token…

### PR DESCRIPTION
**Pull request by aymeric-dispa**
- Store tokens 30 seconds shorter than expiry time to avoid using tokens that are about to expire
- Improve unit tests

### CODE REVIEW

File: src/main/java/com/firebolt/jdbc/service/FireboltAuthenticationService.java [hunk 1]
```diff
@@ -49,6 +50,14 @@ public FireboltConnectionTokens getConnectionTokens(String host, FireboltPropert
 		}
 	}
 
+	/**
+	 * To avoid returning tokens that are about to expire, we store them 30 seconds shorter than their expiry time
+	 */
+	private long getCachingDurationInSeconds(long expireInSeconds) {
+		long thirtySeconds = 30L;
```
Comment (ptiurin) [inline src/main/java/com/firebolt/jdbc/service/FireboltAuthenticationService.java:57]: I'd probably make it a const outside of this function, for visibility.

**Review by ptiurin — COMMENTED** [review]
How did you come up with 30s? What happens if a token expires while a long query is running?

**Comment by aymeric-dispa** [comment]
> How did you come up with 30s? What happens if a token expires while a long query is running?

Usually tokens take hours before expiring. That change will simply make the tokens 'expire' 30 seconds early in the driver so that the client won't face an issue in case the token expires after it has been fetched from the local cache and before the query is sent sent our API. The driver does not cancel any query if a token gets expired (authorization is still performed by the backend - so there is no change regarding long queries). Tokens are fetched every time a new query is sent to our API (so in that regard 30 seconds is actually very safe)

**Review by ptiurin — APPROVED** [review]
apart from that lgtm

**Pull request merged**